src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml

```diff
@@ -1019,6 +1036,9 @@ struct
              ; Params.params.round_constants.(round + 3).(0)
              ; Params.params.round_constants.(round + 3).(1)
              ; Params.params.round_constants.(round + 3).(2)
+             ; Params.params.round_constants.(round + 4).(0)
+             ; Params.params.round_constants.(round + 4).(1)
+             ; Params.params.round_constants.(round + 4).(2)
```

Comment:
wait, am I missing something here? Why aren't the round sorted the same way as the state?

Reply:
They just aren't -- check out the poseidon constraint definition

Reply:
<img width="631" alt="Screen Shot 2021-11-29 at 11 18 16 AM" src="https://user-images.githubusercontent.com/1316043/143929026-770417a4-cb8f-436a-96ae-a98f2c429c00.png">